### PR TITLE
Use versioneer

### LINKED ISSUE #35: Add versioneer
I forgot to increment the `__version__` for the last release. 

I propose switching to [versioneer](https://github.com/warner/python-versioneer) to avoid this in future.

### PR DESCRIPTION
**Pull request by jacobtomlinson**
Closes #35.

In the last few releases I keep forgetting to manually increment `__version__` in `__init__.py` so adding versioneer to do it automatically.

### CODE REVIEW

File: dask_sphinx_theme/__init__.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from os import path
+from . import _version
 
-__version__ = "2.0.3"
+__version__ = _version.get_versions()["version"]
```
Comment (jrbourbeau) [inline dask_sphinx_theme/__init__.py:4]: I see `dask` and `distributed` also define a `__git_revision__` 

https://github.com/dask/dask/blob/aad132a5dad414d987c6358536f41ebf946094b1/dask/__init__.py#L10

I'm not sure exactly where that's used, so I'm not sure if it makes sense to add that attribute here or remove it from `dask` / `distributed`

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1 @@
+dask_sphinx_theme/_version.py export-subst
```
Comment (jrbourbeau) [inline .gitattributes:1]: I'm not sure what this does. Could you provide a little more context?

**Review by jrbourbeau — APPROVED** [review]
Thanks @jacobtomlinson! Looking forward to using this during the next release : ) I left a couple of small, non-blocking comments

File: dask_sphinx_theme/__init__.py (hunk 1 above)
Comment (jrbourbeau) [reply dask_sphinx_theme/__init__.py:4]: Looks like `__git_revision__` is just a nice-to-have which `dask`, `pandas`, and a few other libraries add. I don't think it's a big deal here, but could be added is someone asks for it

File: .gitattributes (hunk 2 above)
Comment (jacobtomlinson) [reply .gitattributes:1]: This is added automatically by versioneer. My understanding is this is the hook that triggers the `dask_sphinx_theme/_version.py` to be replaced with a static version when packaging. I'm not 100% sure how it works but it is automatically added by the `versioneer install` command.

**Pull request merged**